runtime/native/jvm.cpp

```diff
@@ -452,7 +453,9 @@ JVM_GetArrayLength(JNIEnv *env, jobject arr) {
 
 jobject
 JVM_GetArrayElement(JNIEnv *env, jobject arr, jint index) {
-    JvmUnimplemented("JVM_GetArrayElement");
+    // dw475 TODO inspect this
+    return ((jobject*) reinterpret_cast<JArrayRep*>(arr)->Data())[index];
```

Comment:
This array may not be an array of objects (could be an array of primitives).
Mark a TODO for autoboxing primitive types obtained from this (I believe that's what the intended behavior is based on looking at the JDK7 hotspot implementation).